Data/Primitive/ByteArray.hs

```diff
@@ -262,3 +281,49 @@ instance Typeable s => Data (MutableByteArray s) where
   toConstr _ = error "toConstr"
   gunfold _ _ = error "gunfold"
   dataTypeOf _ = mkNoRepType "Data.Primitive.ByteArray.MutableByteArray"
+
+instance Show ByteArray where
+  showsPrec _ ba =
+      showString "ByteArray [" . go 0
```

Comment:
given the `IsList` instance, I think it would be better to have `Show` emit something that would parse on the REPL, i.e.

    [0xca,0xfe,0xbe,0xef]

rather than

    ByteArray [0xca,0xfe,0xbe,0xef]